### PR TITLE
Update docs to latest 3.0.0 build

### PR DESCRIPTION
**Pull request by paul-turner**
Keeping a dev branch for the docs as well to help save work when releasing.
Best to look at this per commit since updating the javadocs added a lot of changes.

### CODE REVIEW

**Comment by kevindekemele** [comment]
Some of the recent updates of 3.0.0 caused some import errors.
I could not import 

import com.bignerdranch.expandablerecyclerview.Model.ParentListItem;
import com.bignerdranch.expandablerecyclerview.Model.ParentWrapper;

After looking in the 3.0.0 branch, it seems the map of 'Model' got a lower case first letter.
So 

import com.bignerdranch.expandablerecyclerview.model.ParentListItem;
import com.bignerdranch.expandablerecyclerview.model.ParentWrapper;

was importable.

**Comment by dgreenhalgh** [comment]
Thanks @kevindekemele. We'll look into this and make sure it's fixed in the snapshot.

File: index.html [hunk 1]
```diff
@@ -108,13 +114,13 @@
                 
     // onCreate ...
     @Override
-    public RecipeViewHolder onCreateParentViewHolder(ViewGroup parentViewGroup) {
+    public RecipeViewHolder onCreateParentViewHolder(ViewGroup parentViewGroup, int viewType) {
         View recipeView = mInflator.inflate(R.layout.recipe_view, parentViewGroup, false);
```
Comment (dgreenhalgh) [inline index.html]: I'll make an issue to fix this typo.

File: index.html [hunk 2]
```diff
@@ -260,20 +266,100 @@
...
+        }
+        return new IngredientViewHolder(ingredientView);
+    }
+                    
+    ...
+}</code></pre>
                 <h2><a id="download" class="anchor" href="#download" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Download</h2>
                 
-                <p><a href="http://repo1.maven.org/maven2/com/bignerdranch/android/expandablerecyclerview/2.1.1/expandablerecyclerview-2.1.1.aar">v2.1.1 AAR</a></p>
+                <p><a href="http://repo1.maven.org/maven2/com/bignerdranch/android/expandablerecyclerview/3.0.0/expandablerecyclerview-2.1.1.aar">v3.0.0 AAR</a></p>
```
Comment (dgreenhalgh) [inline index.html]: 404

File: index.html (hunk 1 above)
Comment (dgreenhalgh) [inline index.html]: #256

File: index.html (hunk 2 above)
Comment (paul-turner) [inline index.html]: Going to have this point at where 3.0.0 would be so it will still 404, but upon release should be fine.

**Pull request merged**